_posts/linda_xiang/2016-12-07-Cloud-data-transfer.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title: "GitHub repository as job scheduling system to orchestrate large data transfer"
+breadcrumb: true
+author: junjun_zhang, linda_xiang
+date: 2016-12-12
+categories: linda_xiang
+tags:
+   - GitHub
+   - data transfer
+   - job scheduling
+teaser:
+   info: We use JSON files to represent data transfer jobs. Each JSON file contains necessary information for a data transfer job. JSON files are pre-generated and checked into a GitHub repository, which is structured with a few directories each representing a job state. Each data transfer worker clones the GitHub repository and picks up a queued job by moving the corresponding JSON file from 'queued-jobs' to 'downloading-jobs' directory. It then commits the change and pushes back to GitHub central repository. If more than one worker picked up the same job, a Git merge conflict will occur which elegantly avoids double scheduling. As the transfer job progresses to next steps, the worker will add log to the job JSON file and move it from one directory to the next, every file change and movement will be committed and pushed to GitHub.
```

Comment:
You might want to abstract this a bit as it seems pretty nuts and bolts. I think focusing on intent in the abstract would help the user have an appreciation for what's to come.